### DISCUSSION TITLE
Checksum mismatch because of synchronous cache rehydration

### DISCUSSION THREAD

**Discussion by mxstbr** [urql-graphql/urql#1319] [Q&A] [answered]
I have a really tricky situation:

- Some of my Next pages are static, some are SSR'd, so I don't want to wrap _app.js with withUrqlClient 
- My `<Navbar />` is rendered from the `<Layout />` component on every page and is fetching the current user. I want that fetch to be client-side on static pages (otherwise they wouldn't be static, duh).
- To do that, I wrapped Navbar in withUrqlClient (separately from the pages) and set `ssr: false`.
- However, now I get a checksum mismatch on SSR'd pages that _also_ (outside of the Navbar) fetch the currentUser! 

I'm relatively certain this happens because the client-side query for the currentUser in the navbar gets synchronously returned from the urql cache, which was rehydrated with the SSR result, which has the currentUser data because another part of the page is also fetching it.

That means on the server (which did not rehydrate from the cache) it rendered the logged-out navbar, but on the very first client-side render it already renders the logged-in navbar. Hence the checksum mismatch.

How do I fix this? The two ideas I can think of:

1. Somehow force the Navbar currentUser query to be async on the first render (`network-only` does not fix this)
2. Even better if possible: SSR the Navbar query on pages that SSR

Are either of these possible or do you have a better idea how to do this?

**Comment by JoviDeCroock** [comment, marked as answer]
Hey,

So if I understand correctly you want to avoid caching user data for CDN caching? There are a few options you got here, one of them being the consideration whether or not that other query is needed (the one causing the user to be in your server-response html).

The second one being to avoid having those authenticated widgets showing in your NavBar from the start.

One that I commonly take is the following:

```jsx
const Navbar = () => {
  const [data] = useQuery({ query: ME_QUERY });
  
   return (
      <nav>
        {...menuItems}
        {isServerSide ? null : renderUserIconIfAuthenticated()}
      </nav>
     )
}
```

**Reply by mxstbr** [reply]
Ahh, that's a smart yet simple way to handle this! Like it, thank you! 👍
